tests/bq_test_helpers.py

```diff
@@ -0,0 +1,96 @@
+"""
+Shared helper functions for BigQuery-related tests
+"""
+import datetime
+import warnings
+from collections import OrderedDict
+from typing import Any, List, Union, Dict
+
+import mock
+from google.cloud import bigquery
+from google.cloud.bigquery import TableReference
+from google.cloud.bigquery.table import TableListItem
+
+_TYPE_TO_FIELD_TYPE = {
+    str(int): 'INT64',
+    str(str): 'STRING',
+    str(float): 'FLOAT64',
+    str(datetime.date): 'DATE',
+    str(datetime.datetime): 'DATETIME'
+}
+dict_types = Union[Dict, OrderedDict]
+
+
+def _field_from_key_value(key: str, value: Any) -> bigquery.SchemaField:
+    """
+    Get a schema field object from a key and value
+    
+    :param key: name of the field
+    :param value: value of the field
+    :return: an appropriate schema field object
+    """
+    tpe = str(type(value))
+    field_type = _TYPE_TO_FIELD_TYPE.get(tpe)
+    if not field_type:
+        raise NotImplementedError(f'The type for {value} is not supported')
+    return bigquery.SchemaField(name=key, field_type=field_type)
+
+
+def fields_from_dict(row):
+    """
+    Get schema fields from a row represented as a dictionary
+
+    :param row: the dictionary to infer schema for
+    :return: list of schema field objects
+    
+    Example:
+        >>> from tests import bq_test_helpers
+        >>> d = {'item': 'book', 'qty': 2, 'price': 1.99}
+        >>> bq_test_helpers.fields_from_dict(d)
+            [SchemaField('item', 'STRING', 'NULLABLE', None, ()),
+             SchemaField('qty', 'INT64', 'NULLABLE', None, ()),
+             SchemaField('price', 'FLOAT64', 'NULLABLE', None, ())]
+    """
+    return [_field_from_key_value(key, value) for key, value in row.items()]
+
+
+def mock_query_result(rows: List[dict_types]):
```

Comment:
@mark-velez require second arg List[str] if first arg is List[dict]

Reply:
We must determine if GCE is 3.7; if so then dict would work just fine and we can avoid the warning altogether.

Reply:
It's still up in the air so I will implement as we discussed above